<launch>

  <arg name="robot" default="firefly_sbx"/>
  <arg name="calibration_file"
    default="$(find orcvio)/config/camchain-imucam-dcist-erl.yaml"/>

  <!-- Image Processor Nodelet  -->
  <group ns="$(arg robot)">
    <node pkg="nodelet" type="nodelet" name="image_processor"
      args="standalone orcvio/ImageProcessorNodelet"
      output="screen">

      <rosparam command="load" file="$(arg calibration_file)"/>

      <param name="grid_row" value="10"/>
      <param name="grid_col" value="5"/>
      <param name="grid_min_feature_num" value="10"/>
      <param name="grid_max_feature_num" value="20"/>
      <param name="pyramid_levels" value="3"/>
      <param name="patch_size" value="15"/>
      <param name="fast_threshold" value="15"/>
      <param name="max_iteration" value="30"/>
      <param name="track_precision" value="0.01"/>
      <param name="ransac_threshold" value="3"/>
      <param name="stereo_threshold" value="5"/>

      <param name="flag_equalize" type="bool"   value="true" />
      <!-- <param name="flag_equalize" type="bool"   value="false" /> -->

      <remap from="~imu" to="/acl_jackal/forward/imu"/>
      <remap from="~cam0_image" to="/acl_jackal/forward/infra1/image_rect_raw"/>
      <remap from="~cam1_image" to="/acl_jackal/forward/infra2/image_rect_raw"/>

    </node>
  </group>

</launch>